src/js/component.js

```diff
@@ -82,1182 +90,1103 @@ var Component = CoreObject.extend({
+   *
+   * @param  {Number|String=} num   Optional width number
+   * @param  {Boolean} skipListeners Skip the 'resize' event trigger
+   * @return {vjs.Component} This component, when setting the width
+   * @return {Number|String} The width, when getting
+   */
+  width(num, skipListeners){
+    return this.dimension('width', num, skipListeners);
+  }
 
-/**
- * Add a CSS class name to the component's element
- *
- * @param {String} classToAdd Classname to add
- * @return {vjs.Component}
- */
-Component.prototype.addClass = function(classToAdd){
-  Lib.addClass(this.el_, classToAdd);
-  return this;
-};
+  /**
+   * Get or set the height of the component (CSS values)
+   *
+   * Setting the video tag dimension values only works with values in pixels.
+   * Percent values will not work.
+   * Some percents can be used, but width()/height() will return the number + %,
+   * not the actual computed width/height.
+   *
+   * @param  {Number|String=} num     New component height
+   * @param  {Boolean=} skipListeners Skip the resize event trigger
+   * @return {vjs.Component} This component, when setting the height
+   * @return {Number|String} The height, when getting
+   */
+  height(num, skipListeners){
+    return this.dimension('height', num, skipListeners);
+  }
 
-/**
- * Remove a CSS class name from the component's element
- *
- * @param {String} classToRemove Classname to remove
- * @return {vjs.Component}
- */
-Component.prototype.removeClass = function(classToRemove){
-  Lib.removeClass(this.el_, classToRemove);
-  return this;
-};
+  /**
+   * Set both width and height at the same time
+   *
+   * @param  {Number|String} width
+   * @param  {Number|String} height
+   * @return {vjs.Component} The component
+   */
+  dimensions(width, height){
+    // Skip resize listeners on width for optimization
+    return this.width(width, true).height(height);
+  }
 
-/**
- * Show the component element if hidden
- *
- * @return {vjs.Component}
- */
-Component.prototype.show = function(){
-  this.removeClass('vjs-hidden');
-  return this;
-};
+  /**
+   * Get or set width or height
+   *
+   * This is the shared code for the width() and height() methods.
+   * All for an integer, integer + 'px' or integer + '%';
+   *
+   * Known issue: Hidden elements officially have a width of 0. We're defaulting
+   * to the style.width value and falling back to computedStyle which has the
+   * hidden element issue. Info, but probably not an efficient fix:
+   * http://www.foliotek.com/devblog/getting-the-width-of-a-hidden-element-with-jquery-using-width/
+   *
+   * @param  {String} widthOrHeight  'width' or 'height'
+   * @param  {Number|String=} num     New dimension
+   * @param  {Boolean=} skipListeners Skip resize event trigger
+   * @return {vjs.Component} The component if a dimension was set
+   * @return {Number|String} The dimension if nothing was set
+   * @private
+   */
+  dimension(widthOrHeight, num, skipListeners){
+    if (num !== undefined) {
```

Comment:
`num` can be set to a default argument of `0`, then we only need to check if it's `null` or `NaN`.